internal/resources/iampolicy/scope.go

```diff
@@ -0,0 +1,240 @@
+	Optional:    true,
+	MaxItems:    1,
+	Elem: &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			orgIDKey: {
+				Type:        schema.TypeString,
+				Description: "ID of the Organization",
+				Required:    true,
+				ForceNew:    true,
+			},
+		},
+	},
+}
+
+var clusterGroupScope = &schema.Schema{
+	Type:        schema.TypeList,
+	Description: "The schema for cluster group iam policy full name",
+	Optional:    true,
+	MaxItems:    1,
+	Elem: &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			clusterGroupNameKey: {
+				Type:        schema.TypeString,
+				Description: "Name of the cluster group",
+				Required:    true,
+				ForceNew:    true,
+			},
+		},
+	},
+}
+
+var clusterScope = &schema.Schema{
+	Type:        schema.TypeList,
+	Description: "The schema for cluster iam policy full name",
+	Optional:    true,
+	MaxItems:    1,
+	Elem: &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			managementClusterNameKey: {
+				Type:        schema.TypeString,
+				Description: "Name of the management cluster",
+				Default:     attachedValue,
+				Optional:    true,
+				ForceNew:    true,
+			},
+			provisionerNameKey: {
+				Type:        schema.TypeString,
+				Description: "Provisioner of the cluster",
+				Default:     attachedValue,
+				Optional:    true,
+				ForceNew:    true,
+			},
+			clusterNameKey: {
+				Type:        schema.TypeString,
+				Description: "Name of the cluster",
+				Required:    true,
+				ForceNew:    true,
+			},
+		},
+	},
+}
+
+var workspaceScope = &schema.Schema{
+	Type:        schema.TypeList,
+	Description: "The schema for workspace iam policy full name",
+	Optional:    true,
+	MaxItems:    1,
+	Elem: &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			workspaceNameKey: {
+				Type:        schema.TypeString,
+				Description: "Name of the workspace",
+				Required:    true,
+				ForceNew:    true,
+			},
+		},
+	},
+}
+
+var namespaceScope = &schema.Schema{
+	Type:        schema.TypeList,
+	Description: "The schema for namespace iam policy full name",
+	Optional:    true,
+	MaxItems:    1,
+	Elem: &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			namespaceNameKey: {
+				Type:        schema.TypeString,
+				Description: "Name of the Namespace",
+				Required:    true,
+				ForceNew:    true,
+			},
+			managementClusterNameKey: {
+				Type:        schema.TypeString,
+				Description: "Name of ManagementCluster",
+				Default:     attachedValue,
+				Optional:    true,
+				ForceNew:    true,
+			},
+			provisionerNameKey: {
+				Type:        schema.TypeString,
+				Description: "Name of Provisioner",
+				Default:     attachedValue,
+				Optional:    true,
+				ForceNew:    true,
+			},
+			clusterNameKey: {
+				Type:        schema.TypeString,
+				Description: "Name of Cluster",
+				Required:    true,
+				ForceNew:    true,
+			},
+		},
+	},
+}
+
+func constructScopeFullName(s scopeType, d *schema.ResourceData) interface{} {
```

Comment:
Can we return a struct with the scope and the full names? Returning an interface requires typecasting to be done, which can be avoided.